ReactWindows/ReactNative/Storage/FutureAccessList.cs

```diff
@@ -0,0 +1,146 @@
+            {
+                StorageApplicationPermissions.FutureAccessList.Remove(token);
+            }
+        }
+
+        /// <summary>
+        /// Resolves a URI returned by the Add(file) method to the file object.
+        /// The file object remains in the cache and can be retrieved again later.
+        /// </summary>
+        public static async Task<StorageFile> GetFileAsync(string uri)
+        {
+            var token = ExtractAccessToken(uri);
+            return await StorageApplicationPermissions.FutureAccessList.GetFileAsync(token);
+        }
+
+        /// <summary>
+        /// Resolves a URI returned by the Add(folder) method to the folder object.
+        /// The folder object remains in the cache and can be retrieved again later.
+        /// </summary>
+        public static async Task<StorageFolder> GetFolderAsync(string uri)
+        {
+            var token = ExtractAccessToken(uri);
+            return await StorageApplicationPermissions.FutureAccessList.GetFolderAsync(token);
+        }
+
+        /// <summary>
+        /// Checks if FAL contains a file or folder with the given URI.
+        /// Returns false If the URI is invalid.
+        /// </summary>
+        public static bool ContainsItem(string uri)
+        {
+            if (TryExtractAccessToken(uri, out var token))
+            {
+                return StorageApplicationPermissions.FutureAccessList.ContainsItem(token);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private static bool TryExtractAccessToken(string uri, out string token)
+        {
+            if (!uri.StartsWith(Scheme))
+            {
+                token = null;
+                return false;
+            }
+            else
+            {
+                token = Uri.UnescapeDataString(uri.Substring(Scheme.Length));
+                return true;
+            }
+        }
+
+        private static string ExtractAccessToken(string uri)
+        {
+            if (!TryExtractAccessToken(uri, out var token))
+            {
+                throw new ArgumentException("Invalid future access list URI: " + uri);
+            }
+            else
+            {
+                return token;
+            }
+        }
+
+        private static string GetAccessToken(IStorageItem item)
+        {
+            var futureAccessList = StorageApplicationPermissions.FutureAccessList;
+            var existingEntries = futureAccessList.Entries;
+            var capacity = futureAccessList.MaximumItemsAllowed;
+            var now = DateTime.UtcNow;
+
+            var expiry = existingEntries.Count >= capacity / 2 ?
+                now.AddDays(-1) :
+                now.AddDays(-7);
+
+            foreach (var entry in existingEntries)
+            {
+                if (long.TryParse(entry.Metadata, out long ticks))
+                {
+                    var time = new DateTime(ticks);
```

Comment:
>DateTime [](start = 35, length = 8)

Is this guaranteed to be a UTC `DateTime`? You use `UtcNow` above.

Reply:
Indeed. There is second param that can be `DateTimeKind.Utc` or something else. I mistakenly assumed that the C#'s `new DateTime(123)` behaves like the JS's `new Date(123)`.